composition/content_merge.go

```diff
@@ -49,59 +53,90 @@ func NewContentMerge(metaJSON map[string]interface{}) *ContentMerge {
 	return cntx
 }
 
-func (cntx *ContentMerge) GetHtml() ([]byte, error) {
-	if len(cntx.priorities) > 0 {
-		cntx.processMetaPriorityParsing()
+func (cntx *ContentMerge) collectStylesheets(f Fragment) {
+	cntx.stylesheets = append(cntx.stylesheets, f.Stylesheets()...)
+}
+
+func (cntx *ContentMerge) writeStylesheets(w io.Writer) {
+	for _, href := range cntx.stylesheets {
+		stylesheet := fmt.Sprintf("\n    <link rel=\"stylesheet\" type=\"text/css\" href=\"%s\">", href)
+		io.WriteString(w, stylesheet)
 	}
-	w := bytes.NewBuffer(make([]byte, 0, DefaultBufferSize))
+}
 
-	var executeFragment func(fragmentName string) error
+func generateExecutionFunction(cntx *ContentMerge, w io.Writer) (executeFragment func(fragmentName string) error) {
 	executeFragment = func(fragmentName string) error {
 		f, exist := cntx.GetBodyFragmentByName(fragmentName)
 		if !exist {
 			missingFragmentString := generateMissingFragmentString(cntx.Body, fragmentName)
 			return errors.New(missingFragmentString)
 		}
+		cntx.collectStylesheets(f)
 		return f.Execute(w, cntx.MetaJSON, executeFragment)
 	}
+	return executeFragment
+}
 
-	io.WriteString(w, "<!DOCTYPE html>\n<html>\n  <head>\n    ")
+func (cntx *ContentMerge) GetHtml() ([]byte, error) {
+
+	if len(cntx.priorities) > 0 {
+		cntx.processMetaPriorityParsing()
+	}
+
+	// start header, but don't close it. We will add stylsheets later on
+	header := bytes.NewBuffer(make([]byte, 0, DefaultBufferSize))
+	io.WriteString(header, "<!DOCTYPE html>\n<html>\n  <head>\n    ")
 
 	for _, f := range cntx.Head {
-		if err := f.Execute(w, cntx.MetaJSON, executeFragment); err != nil {
+		cntx.collectStylesheets(f)
```

Comment:
The collectStylesheets(f) call within the closure is executed against nested fragments. Here, the stylesheets of the parent fragment are collected.